### PR TITLE
Basic image processing based workflow/detection

### PR DESCRIPTION
**Pull request by jorisvandenbossche**

### CODE REVIEW

File: problem.py [hunk 1]
```diff
@@ -1,46 +1,56 @@
...
-    folder = os.path.join(path, 'data', 'imgs')
-    return (folder, X), y
+    src = np.load('data/images_quad_77.npy', mmap_mode='r')
+    labels = pd.read_csv("data/quad77_labels.csv")
+
+    y = [list(labels[labels.id == '77_{0}'.format(i)][
+                  ['x_p', 'y_p', 'radius_p']].itertuples(name=None,
+                                                         index=False))
+         for i in range(src.shape[0])]
+
```
Comment (aboucaud) [inline problem.py]: It took me 30 sec to understand how the `y` is actually created 😄 
Maybe add a tiny bit of explanation for a non pandas-ninja..

File: submissions/image_processing/batch_classifier.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        return self
+
+    def _hough_detection(self, image):
+        edges = canny(image, sigma=self.sigma, low_threshold=10,
+                      high_threshold=50)
+
+        hough_radii = list(np.arange(10, 20, 1)) + list(
+            np.arange(20, 40, 2)) + list(np.arange(40, 100, 4))
+
+
```
Comment (aboucaud) [inline submissions/image_processing/batch_classifier.py]: So in the end we are using the radii from 10 to 100, not 5 to 50 ?

File: workflow/local_workflow.py [hunk 3]
```diff
@@ -268,45 +259,26 @@ def _chunk_iterator(X_array, folder, y_array=None, chunk_size=1024, n_jobs=8,
...
+        # X = Parallel(n_jobs=n_jobs, backend='threading')(delayed(imread)(
+        #    filename) for filename in filenames)
+        X = src[i:i + chunk_size, :, :]
+
+        if labels is not None:
+            # y = y_array[i:i + chunk_size]
+            y = [list(labels[labels.id == '77_{0}'.format(j)][
+                          ['x_p', 'y_p', 'radius_p']].itertuples(name=None,
+                                                                 index=False))
+                 for j in range(i, i + chunk_size)]
```
Comment (aboucaud) [inline workflow/local_workflow.py]: Same comment as previous

File: submissions/image_processing/batch_classifier.py (hunk 2 above)
Comment (jorisvandenbossche) [reply submissions/image_processing/batch_classifier.py]: ah, good catch. No, I just copy pasted the old, wrong values

File: workflow/local_workflow.py (hunk 3 above)
Comment (jorisvandenbossche) [reply workflow/local_workflow.py]: This can actually be removed now, but will clarify it in the other occurence

**Comment by jorisvandenbossche** [comment]
I am going to merge this, so we can further work on this basic workflow for other models / score types.

(the only problem for testing out is that the current model is *very* bad (aggregated OSPA of 0.002 on the first 100 patches), so the trying out different scores won't give sensible results)

**Comment by kegl** [comment]
A simple NN could be one that is trained on only images containing a single crater to regress x, y, r. Or if there are two many images with no craters, first a classifier whether there is a crater, fed to the one that detects exactly one.

**Pull request merged**